File: 01Singleton/Students/2018/MielewczykLukasz/singleton.py
import threading

class Singleton(object):
	threadingLock = threading.Lock()
	instance = None
	
	def __new__(self):
		if not self.instance:
			with Singleton.threadingLock:
				if not self.instance:
					self.instance = object.__new__(self)
		return self.instance

class SubSingleton(Singleton):
	def __new__(self):
		return super(SubSingleton, self).__new__(self)

File: 01Singleton/Students/2018/Singleton.php
<?php

namespace App;


class Singleton
{
    protected static $instance;

    public static function getInstance()
    {
        if (self::$instance === null) {
            self::$instance = new Singleton();
        }

        return self::$instance;
    }
}

class SubSingleton extends Singleton
{
    protected static $instance;

    public static function getInstance()
    {
        if (self::$instance === null) {
            self::$instance = new SubSingleton();
        }

        return self::$instance;
    }
}

$singleton = Singleton::getInstance();
$singleton2 = SubSingleton::getInstance();

print_r($singleton);
print_r($singleton2);